filer/admin/fileadmin.py

```diff
@@ -174,10 +175,13 @@ def icon_view(self, request, file_id: int, size: int) -> HttpResponse:
         if not isinstance(file, BaseImage):
             raise Http404()
 
-        thumbnailer = get_thumbnailer(file)
-        thumbnail_options = ThumbnailOptions({'size': (size, size), "crop": True})
-        thumbnail = thumbnailer.get_thumbnail(thumbnail_options, generate=True)
-        return HttpResponseRedirect(thumbnail.url)
+        try:
+            thumbnailer = get_thumbnailer(file)
+            thumbnail_options = ThumbnailOptions({'size': (size, size), "crop": True})
+            thumbnail = thumbnailer.get_thumbnail(thumbnail_options, generate=True)
+            return HttpResponseRedirect(thumbnail.url)
+        except InvalidImageFormatError:
+            raise Http404
```

Comment:
```suggestion
            raise Http404
            return HttpResponseRedirect(staticfiles_storage.url('filer/icons/file-missing.svg'))
```
This would render the correct icon for files currently not accessible. 
![file-missing](https://github.com/django-cms/django-filer/assets/16904477/ab4ba43f-4d5c-4b1a-b103-6e08af6cdb0a)

Reply:
@vinitkumar You'd also have to add `from django.contrib.staticfiles.storage import staticfiles_storage` at the top of the file...